components/script/textinput.rs

```diff
@@ -492,15 +490,20 @@ impl<T: ClipboardProvider> TextInput<T> {
                             [..remaining];
                     }
 
-                    match input
-                        .unicode_words()
-                        .last() {
-                            None => 0,
-                            Some(i) => {
-                                - (input.len() as isize + newline_adjustment as isize
-                                - input.rfind(i).expect("no words to the left on this line") as isize)
-                            },
-                       }
+                    let mut iter = input.split_word_bounds().rev();
+                    let mut shift_temp: isize = 0;
+                    loop {
+                        match iter.next() {
+                            None => break,
+                            Some(x) => {
+                                shift_temp += - (x.len() as isize);
+                                if x.chars().any(|x| x.is_alphabetic() || x.is_numeric()) {
```

Comment:
The `unicode-segmentation` crate (using `.split_word_bounds()`) determines word boundaries differently than how it should be implemented for this keyboard shortcut.

For example, "hello .. world" would be split up as ["hello", " ", ".", ".", " ","world"], whereas if the cursor was for example at the end of "hello", you would want this function to jump to the end of the "world" word. (
> after splitting the string on word boundaries, contain any characters with the Alphabetic property, or with General_Category=Number.
)